### PR TITLE
make primitive creation aware of reorder_src/reorder_weight setting

### PR DESCRIPTION
**Pull request by XiaobingSuper**
As the title, we should make primitive creation aware of the reorder_src/reorder_weight setting.

### CODE REVIEW

**Comment by XiaobingSuper** [comment]
@jgong5, @[EikanWang](https://github.com/EikanWang), please help review this PR.

File: include/ideep/operators/inner_product.hpp [hunk 1]
```diff
@@ -526,7 +537,11 @@ struct inner_product_forward
       } else {
         dst.feed_from(expected_dst);
       }
-    } else { // reorder_src
+    } else {
+      if (!reorder_src)  {
```
Comment (jgong5) [inline include/ideep/operators/inner_product.hpp:541]: This check is not necessary?

File: include/ideep/operators/inner_product.hpp [hunk 2]
```diff
@@ -499,7 +510,7 @@ struct inner_product_forward
     args.insert({DNNL_ARG_SCRATCHPAD, scratchpad});
     args.insert(
         {DNNL_ARG_ATTR_MULTIPLE_POST_OP(0) | DNNL_ARG_SRC_1, expected_other});
-    if (reorder_src) {
+    if (reorder_src) { // reorder src
```
Comment (jgong5) [inline include/ideep/operators/inner_product.hpp]: keep "// reorder src" comment at the "else" clause?

File: include/ideep/operators/inner_product.hpp [hunk 3]
```diff
@@ -597,7 +612,11 @@ struct inner_product_forward
       } else {
         dst.feed_from(expected_dst);
       }
-    } else { // reorder_src
+    } else {
+      if (!reorder_src)  {
```
Comment (jgong5) [inline include/ideep/operators/inner_product.hpp:616]: ditto.

**Review by jgong5 — APPROVED** [review]
LGTM except for some small nits. Does the "matmul" primitive have sijmilar issue?

**Review by EikanWang — APPROVED** [review]

File: include/ideep/operators/inner_product.hpp (hunk 1 above)
Comment (XiaobingSuper) [reply include/ideep/operators/inner_product.hpp:541]: I think it is necessary, which make sure the user gives a dst(will use it's format ) when reorder_src is false.

Comment (XiaobingSuper) [reply include/ideep/operators/inner_product.hpp:541]: > LGTM except for some small nits. Does the "matmul" primitive have sijmilar issue?

matmul doesn't have such an issue, it always uses input's desc to create primitive.

File: include/ideep/operators/inner_product.hpp (hunk 2 above)
Comment (XiaobingSuper) [reply include/ideep/operators/inner_product.hpp]: changed.

**Pull request merged**